src/utils/payloadStringify.js

```diff
@@ -32,18 +32,27 @@ const keyToRegexes = (
   backwardCompRegexEnvVarName = LUMIGO_SECRET_MASKING_REGEX_BACKWARD_COMP,
   regexesEnvVarName = LUMIGO_SECRET_MASKING_REGEX
 ) => {
-  if (process.env[backwardCompRegexEnvVarName]) {
-    const parseResponse = parseJsonFromEnvVar(backwardCompRegexEnvVarName, true);
-    if (parseResponse) {
-      regexesList = parseResponse;
-    }
-  } else if (process.env[regexesEnvVarName]) {
-    const parseResponse = parseJsonFromEnvVar(regexesEnvVarName, true);
-    if (parseResponse) {
-      regexesList = parseResponse;
+  const fallbackRegexesList = regexesList;
+
+  const tryParseEnvVar = (envVarName) => {
+    if (process.env[envVarName]) {
+      return parseJsonFromEnvVar(envVarName, true);
     }
+    return null;
+  };
+
+  // Try parsing backward compatibility or main environment variables
+  const regexes =
+    tryParseEnvVar(backwardCompRegexEnvVarName) || tryParseEnvVar(regexesEnvVarName) || regexesList;
+
+  try {
+    return regexes.map((x) => new RegExp(x, 'i'));
+  } catch (e) {
+    regexes.forEach((x) =>
+      invalidMaskingRegexWarning(`Error processing regex: ${x} - ${e.message}`)
+    );
+    return fallbackRegexesList.map((x) => new RegExp(x, 'i'));
```

Comment:
This will print a warning for all regexes, even for valid one, if a one or more in the list are invalid.
I suggest to use something like this:
```js
const parsedRegexes = regexes.map((x) => {
  try {
    return new RegExp(x, 'i')
  } catch (e) {
    invalidMaskingRegexWarning(`Error processing regex: ${x} - ${e.message}`)
    return e;
  }
});

if (parsedRegexes.some((x) => x instanceof Error)) {
  // print warning about using fallbacks
} else {
  return parsedRegexes
}
```
and print a message about using fallbacks, so it will be easier for the user to understand that the defaults patterns were used instead.

Reply:
I am about to use the same as in the `shallowMask` method

```
 try {
      givenSecretRegexes = JSON.parse(givenSecretRegexes);
      givenSecretRegexes = givenSecretRegexes.map((x) => new RegExp(x, 'i'));
    } catch (e) {
      invalidMaskingRegexWarning(e);
      givenSecretRegexes = null;
    }
```

I have it changed to

```
  try {
    return regexes.map((x) => new RegExp(x, 'i'));
  } catch (e) {
    invalidMaskingRegexWarning(e);
    logger.warn('Fallback to default regexes list', { fallbackRegexesList });
    return fallbackRegexesList.map((x) => new RegExp(x, 'i'));
  }
```

Reply:
Not sure I understand. What I meant is that we would like to print the regex value that was invalid, instead of letting the user look it up manually.
In the latter version it's still not printing the exact item in the list that's invalid, but just a generic message...